Migrate creators even when dublin core behaviors are not enabled.

jone opened the pull request:
The Dexterity base class has support for creators (it implements listCreators() and other methods. The dublin core behavior makes it possible to edit the creators.

The migration did work when the dublin core behavior was enabled. But when it was not enabled, the creators was overwritten by the user executing the migration. This change updates the creators.

busykoala commented on ftw/upgrade/tests/test_migration.py:
```diff
@@ -383,7 +384,37 @@ def test_migrate_ownership(self):
+
+        # The creators list behaves differently when the dublin core
+        # behavior is used.
+        self.portal.portal_types['Folder'].behaviors = tuple(
```
Do I understand that correctly, that here the behaviour IDublinCore is getting removed from all folder? If thats the case an assertion would maybe help to understand that it is important to have the behaviour removed on folders.

jone replied:
yes: the goal of the test is to verify that the creators are migrated properly when the bahavior is not activated on the target type. That's why in the setup of the test we are removing the behavior from the FTI. But it is not the goal that the behavior is removed.
The change is also restored after this test.

djowett-ftw commented on ftw/upgrade/tests/base.py:
```diff
@@ -50,6 +53,65 @@ def grant(self, *roles):
+
+        :param user: A user object or a user name.
+        :type user: string or user object
+        :param browser: A ``ftw.testbrowser`` instance to log in too.
```
`login to` or `log in to` or `log into` - but def not `too`.  I prefer `log into` (as does Facebook)

https://english.stackexchange.com/questions/5302/log-in-to-or-log-into-or-login-to

jone replied:
😂

djowett-ftw commented:
One more qu: Why `def login(...)` here rather than in (say) ftw.testing?

jone commented:
> One more qu: Why `def login(...)` here rather than in (say) ftw.testing?

I have actually copied it from `ftw.testing`. I need to improve that at some point: in `ftw.testing` the `login()` implementation is in an integration test case base class, but we are using functional testing here. I think we should have a functional test case base class at some point too, but for now it's easier to just put it here, I think 🤔

djowett-ftw commented:
Ok, understood - maybe strip out most of the docstring to avoid it looking too "supported"?

jone commented:
> Ok, understood - maybe strip out most of the docstring to avoid it looking too "supported"?

👍 good idea, I've removed the docstring.
I've also added some `transaction.commit` statements because that's needed in functional testing (my test just does not use a browser in practice, that's why it does not break something)

jone commented:
👍 updated
